src/components/WithBackgroundImage/index.js

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+
+import useBaseUrl from '@docusaurus/useBaseUrl';
+import useDocusaurusContext from '@docusaurus/useDocusaurusContext';
+import useThemeContext from '@theme/hooks/useThemeContext';
+
+const WithBackgroundImage = ({
```

Comment:
This is a HOC to handle the dark mode toggle logic. And this should be faster, because only the smaller component (this) will rerender instead of the a larger component (such as the whole sidebar)